cmd/cli/version/version.go

```diff
@@ -104,13 +107,71 @@ func (oV *VersionOptions) Run(ctx context.Context, cmd *cobra.Command) error {
 	if !oV.ClientOnly {
 		serverVersion, err := util.GetAPIClient(ctx).Version(ctx)
 		if err != nil {
-			log.Ctx(ctx).Error().Err(err).Msgf("could not get server version")
-			return err
+			return fmt.Errorf("error running version: %w", err)
+		} else {
+			versions.ServerVersion = serverVersion
+			columns = append(columns, serverVersionColumn)
 		}
+	}
+
+	// Add additional information
+	versions.OperatingSystem = runtime.GOOS
+	versions.Architecture = runtime.GOARCH
+	versions.UserID = "some-user-id" // Replace this with your method to get UserID
 
-		versions.ServerVersion = serverVersion
-		columns = append(columns, serverVersionColumn)
+	var serverVersion string
+	if versions.ServerVersion != nil {
+		serverVersion = versions.ServerVersion.GitVersion
 	}
 
+	checkForUpdates(ctx, versions.ClientVersion.GitVersion, serverVersion, versions.OperatingSystem, versions.Architecture, versions.UserID)
+
 	return output.OutputOne(cmd, columns, oV.OutputOpts, versions)
 }
+
+type ServerResponse struct {
+	Version string `json:"version"`
+	Message string `json:"message"`
+}
+
+func checkForUpdates(ctx context.Context, currentClientVersion, currentServerVersion, os, arch, userID string) {
+	u, err := url.Parse("http://update.bacalhau.org/version")
```

Comment:
Where does the code live that backs this endpoint, or is this being served by our public requester node?

We should probably make endpoint part of the config file, see `pkg/config`. 
 - This will make testing easier as we could replace the endpoint with a stubbed local version while testing this command. 
 - It's also not guaranteed that all bacalhau networks run the same version, e.g. staging and production are on different version - v1.0.4 and v1.0.3 respectively.

Reply:
Also, we need to be sure not to hard-code `http` for the URL once the host is configured.